tests/Form/Type/AdminTypeTest.php

```diff
@@ -127,6 +132,91 @@ public function testDotFields(): void
         }
     }
 
+    public function testArrayCollection(): void
+    {
+        $foo = new Foo();
+
+        $parentSubject = new \stdClass();
+        $parentSubject->foo = new ArrayCollection([$foo]);
+
+        $parentAdmin = $this->prophesize(AdminInterface::class);
+        $parentAdmin->getSubject()->shouldBeCalled()->willReturn($parentSubject);
+        $parentAdmin->hasSubject()->shouldBeCalled()->willReturn(true);
+        $parentField = $this->prophesize(FieldDescriptionInterface::class);
+        $parentField->getAdmin()->shouldBeCalled()->willReturn($parentAdmin->reveal());
+
+        $modelManager = $this->prophesize(ModelManagerInterface::class);
+
+        $admin = $this->prophesize(AbstractAdmin::class);
+        $admin->hasParentFieldDescription()->shouldBeCalled()->willReturn(false);
+        $admin->getParentFieldDescription()->shouldBeCalled()->willReturn($parentField->reveal());
+        $admin->defineFormBuilder(new AnyValueToken())->shouldBeCalled();
+        $admin->getModelManager()->shouldBeCalled()->willReturn($modelManager);
+        $admin->getClass()->shouldBeCalled()->willReturn(Foo::class);
+        $admin->setSubject($foo)->shouldBeCalled();
+
+        $field = $this->prophesize(FieldDescriptionInterface::class);
+        $field->getAssociationAdmin()->shouldBeCalled()->willReturn($admin->reveal());
+        $field->getFieldName()->shouldBeCalled()->willReturn('foo');
+        $field->getParentAssociationMappings()->shouldBeCalled()->willReturn([]);
+
+        $this->builder->add('foo');
+
+        $type = new AdminType();
+        try {
+            $type->buildForm($this->builder, [
+                'sonata_field_description' => $field->reveal(),
+                'delete' => false, // not needed
+                'property_path' => '[0]', // actual test case
+            ]);
+        } catch (NoSuchPropertyException $exception) {
+            $this->fail($exception->getMessage());
+        }
+    }
+
+    public function testArrayCollectionNotFound(): void
+    {
+        $parentSubject = new \stdClass();
+        $parentSubject->foo = new ArrayCollection();
+
+        $parentAdmin = $this->prophesize(AdminInterface::class);
+        $parentAdmin->getSubject()->shouldBeCalled()->willReturn($parentSubject);
+        $parentAdmin->hasSubject()->shouldBeCalled()->willReturn(true);
+        $parentField = $this->prophesize(FieldDescriptionInterface::class);
+        $parentField->getAdmin()->shouldBeCalled()->willReturn($parentAdmin->reveal());
+
+        $modelManager = $this->prophesize(ModelManagerInterface::class);
+
+        $foo = new Foo();
+
+        $admin = $this->prophesize(AbstractAdmin::class);
+        $admin->hasParentFieldDescription()->shouldBeCalled()->willReturn(false);
+        $admin->getParentFieldDescription()->shouldBeCalled()->willReturn($parentField->reveal());
```

Comment:
If call to `hasParentFieldDescription()` will return `false`, should not `getParentFieldDescription()` return `null`?

Reply:
Most of the test is copied from previous one.

Funny thing. If getParentFieldDescription return null the AdminType crash.
It was never supported to use this form with no parentFieldDescription.

Reply:
I added consistency in the return value of the mock in the test.